cilium/cmd/debuginfo.go

```diff
@@ -77,14 +96,82 @@ var sections = map[string]addSection{
 
 func init() {
 	rootCmd.AddCommand(debuginfoCmd)
-	debuginfoCmd.Flags().StringVarP(&file, "file", "f", "", "Redirect output to file")
-	debuginfoCmd.Flags().StringVarP(&html, "html-file", "", "", "Convert default output to HTML file")
+	debuginfoCmd.Flags().BoolVarP(&outputToFile, "file", "f", false, "Redirect output to file(s)")
 	debuginfoCmd.Flags().BoolVarP(&filePerCommand, "file-per-command", "", false, "Generate a single file per command")
+	debuginfoCmd.Flags().StringSliceVar(&outputOpts, "output", []string{}, "markdown| html| json| jsonpath='{}'")
+	debuginfoCmd.Flags().StringVar(&outputDir, "output-directory", "", "directory for files (if specified will use directory in which this command was ran)")
+}
+
+func validateInput() {
+	if outputDir != "" && !outputToFile {
+		fmt.Fprintf(os.Stderr, "invalid option combination; specified output-directory %q, but did not specify for output to be redirected to file; exiting\n", outputDir)
+		os.Exit(1)
+	}
+	validateOutputOpts()
+}
+
+func validateOutputOpts() {
+
+	for _, outputOpt := range outputOpts {
+		switch strings.ToUpper(outputOpt) {
+		case markdownOutput:
+		case htmlOutput:
+			if !outputToFile {
+				fmt.Fprintf(os.Stderr, "if HTML is specified as the output format, it is required that you provide the `--file` argument as well\n")
+				os.Exit(1)
+			}
+		case jsonOutputDebuginfo:
+			if filePerCommand {
+				fmt.Fprintf(os.Stderr, "%s does not support dumping a file per command; exiting\n", jsonOutputDebuginfo)
+				os.Exit(1)
+			}
+		case jsonpathOutput:
```

Comment:
FWIW I think that if you used `outputOpt` or `strings.ToUpper(outputOpt)` as the argument to the error line, then you could fold the `jsonOutputDebuginfo` and `jsonpathOutput` cases together here:

```suggestion
		case jsonOutputDebuginfo, jsonpathOutput:
￼ 			if filePerCommand {
				fmt.Fprintf(os.Stderr, "%s does not support dumping a file per command; exiting\n", outputOpt)
```